site/README.md

```diff
@@ -0,0 +1,57 @@
+# Netlify CLI docs site
+
+[![Built with Starlight](https://astro.badg.es/v2/built-with-starlight/tiny.svg)](https://starlight.astro.build)
+
+## Architecture
+
+All the command reference pages and the list of commands are automatically generated from the CLI code. This is
+automatically inserted into the existing, committed, semi-manually-written pages in `../docs/`. Here's how the flow
+works:
+
+1. The `docs.js` script replaces the content between the auto-generation marker comments in the files in `../docs/`; the
+   rest you can update manually.
+2. The `sync.js` script copies these over to this Astro Starlight site, into `src/content/docs/`. It may also perform
+   some minor transformations.
+3. From there, it's any old Astro/Starlight docs site, as documented.
```

Comment:
😅 This doesn't seem that complex but it took me a lot of trial-and-error and poking around to understand the whole flow. I hope this documentation is clear for the next person.